src/Microsoft.Extensions.Logging.Testing/Xunit/LoggedTestInvoker.cs

```diff
@@ -0,0 +1,39 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Reflection;
+using System.Threading;
+using Xunit.Abstractions;
+using Xunit.Sdk;
+
+namespace Microsoft.Extensions.Logging.Testing
+{
+    public class LoggedTestInvoker : XunitTestInvoker
+    {
+        public LoggedTestInvoker(ITest test, IMessageBus messageBus, Type testClass, object[] constructorArguments, MethodInfo testMethod, object[] testMethodArguments, IReadOnlyList<BeforeAfterTestAttribute> beforeAfterAttributes, ExceptionAggregator aggregator, CancellationTokenSource cancellationTokenSource) : base(test, messageBus, testClass, constructorArguments, testMethod, testMethodArguments, beforeAfterAttributes, aggregator, cancellationTokenSource)
+        {
+        }
+
+        protected override object CreateTestClass()
+        {
+            var testClass = base.CreateTestClass();
+
+            if (testClass is LoggedTest loggedTestClass)
+            {
+                var classType = loggedTestClass.GetType();
+                var testOutputHelper = ConstructorArguments.Single(a => typeof(ITestOutputHelper).IsAssignableFrom(a.GetType())) as ITestOutputHelper;
+                var logLevelAttribute = TestMethod.GetCustomAttribute(typeof(LogLevelAttribute)) as LogLevelAttribute;
+                var testName = TestMethodArguments.Aggregate(TestMethod.Name, (a, b) => $"{a}_{(b ?? "null")}");
```

Comment:
`string.Join('_', ...)`?

Reply:
Closest I can come up with is `string.Join("_", TestMethod.Name, TestMethodArguments)` but that won't be able to mark `null` object with `"null"` string

Reply:
From adding tests previously, I remember it being useful to distinguish between null strings and empty strings for example. That's why I special cased null here.

Reply:
👍 for foldl

Reply:
What is foldl?

Reply:
The hipster name for Aggregate.